src/spatialdata/_core/query/relational_query.py

```diff
@@ -448,19 +539,53 @@ def match_table_to_element(sdata: SpatialData, element_name: str) -> AnnData:
     sdata
         SpatialData object
     element_name
-        Name of the element to match the table to
+        The name of the spatial elements to be joined with the table.
 
     Returns
     -------
     Table with the rows matching the instances of the element
     """
+    # TODO: refactor this to make use of the new join_sdata_spatialelement_table function.
```

Comment:
Probably the easiest will be to replace the existing calls of `match_table_to_element()` with `join_sdata_spatialelement_table()`.